ui/pages/confirmations/hooks/useConfirmationAlertMetrics.ts

```diff
@@ -117,7 +113,17 @@ export function useConfirmationAlertMetrics() {
     if (!properties) {
       return;
     }
-    updateTransactionEventFragment({ properties }, ownerId);
+
+    if (isSignatureTransactionType(currentConfirmation)) {
+      const requestId = (currentConfirmation as SignatureRequestType)?.msgParams
+        ?.requestId;
+      const fragmentUniqueId = `signature-${requestId}`;
+      updateEventFragment(fragmentUniqueId, {
```

Comment:
Is it worth a parallel `useSignatureEventFragment` hook to encapsulate the generation of the fragment ID and allow other parts of the UI to add properties to signatures?

Reply:
I'm thinking yes. Though we don't yet have additional use-cases yet, I think it's likely we will later. Working on including this directly into this PR. Could also complete in separate PR if this PR gets approved x2 and merged

Reply:
hey @matthewwalsh0! I think this is a good idea. We don't have additional use-cases for this yet, but I imagine we will. I have some code written now, but I think it should go in a separate PR. Will create one following the merge of this one

Reply:
or something similar to `updateTransactionEventFragment` to encapsulate this whole thing.